src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -89,40 +104,81 @@ const DotPlotPage = (props) => {
     if (hierarchy.length === 0) dispatch(loadCellSets(experimentId));
   }, []);
 
-  useEffect(() => {
-    if (config && config.selectedGenes.length === 0 && !genesFetching && !highestDispersionGenes) {
-      const PROPERTIES = ['dispersions'];
-      const tableState = {
-        pagination: {
-          current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
-        },
-        geneNamesFilter: null,
-        sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
-      };
+  const currentConfig = useRef(null);
+
+  const getCurrentDataProps = (conf) => ({
```

Comment:
why do we need this function? what extra fields does `conf` contain that are not needed? do they need to be removed?

Reply:
This is a helper function to make it easy to get the props that defines the plot's data. The idea is we want the plot to re-fetch the plot's data if these data changes. If other plot config changes, we want the plot to update, but not refetch the data.